### ISSUE TITLE
Build a new version of Bigtable library

### ISSUE THREAD

**Issue by mbrukman** [OpenTSDB/asyncbigtable#20]
Hi @csoulios, since you've updated the version of the Bigtable client library in to address issue #16, could you please build a new version of the library and update OpenTSDB library deps accordingly?

The latest version on Maven Central is currently 0.1.0 from Oct 2015, which is quite out-of-date.

Thanks!

**Comment by csoulios** [comment]
Hi @mbrukman, I plan to work on this next week. I want to fix travis builds first, then do some tests and finally release it by end of next week. 

Thanks

**Comment by mbrukman** [comment]
Hi @csoulios, any updates on the new release?

With your upcoming talk on OpenTSDB + Bigtable at Percona Live, it would be good to have it out by then so that folks try out OpenTSDB with a high-performance Bigtable client.

**Comment by goll** [comment]
+1
A couple of questions:
* Is `netty-tcnative-boringssl-static` required in pom.xml as referenced [here](https://github.com/GoogleCloudPlatform/cloud-bigtable-client#using-the-java-client)?
* Shouldn't [this](https://github.com/OpenTSDB/asyncbigtable/blob/master/pom.xml#L225) be `bigtable-hbase-1-2` since that's the latest version?
* Can any of the alpha/beta deps (protobuf-java, netty-all) in pom.xml be bumped to stable versions?
* [This dep](https://github.com/OpenTSDB/asyncbigtable/blob/master/pom.xml#L275) has already been bumped upstream: https://github.com/OpenTSDB/opentsdb/pull/927

**Comment by mbrukman** [comment]
@goll:

> * Shouldn't this be bigtable-hbase-1-2 since that's the latest version?

The version number is not the version of the package, but the version of HBase that it's compatible with. So if you're using HBase 1.0, you would use `bigtable-hbase-1.0` and similarly for HBase 1.1 and 1.2 releases. More info in the docs: [Bigtable and HBase](https://cloud.google.com/bigtable/docs/bigtable-and-hbase) and [connecting to Bigtable using HBase API](https://cloud.google.com/bigtable/docs/connecting-hbase#creating_the_hbase-sitexml_file), though I can see how it might be confusing.

@sduskis and @igorbernstein2 — what are your thoughts on whether the dependency should be using 1.2 here instead of 1.1? [`cloud-bigtable-client` README](https://github.com/GoogleCloudPlatform/cloud-bigtable-client#using-the-java-client) recommends version 1.2 by default, FWIW.

**Comment by goll** [comment]
FYI I have tested a custom built asyncbigtable by adding the following to `pom.xml` and using it in OpenTSDB 2.3.0:
```
    <dependency>
      <groupId>com.google.cloud.bigtable</groupId>
      <artifactId>bigtable-hbase-1.2</artifactId>
      <version>0.9.4</version>
    </dependency>

    <dependency>
      <groupId>io.netty</groupId>
      <artifactId>netty-tcnative-boringssl-static</artifactId>
      <version>1.1.33.Fork19</version>
    </dependency>
```
Running without issues on GCP multiple Bigtable instance.

**Comment by csoulios** [comment]
Hi @mbrukman and @goll 

I have completed the version upgrades of bigtable-hbase-1.2 to 0.9.6, protobuf-java to 3.0.2, async 1.4.1 and netty-all to 4.1.0.Final. The first tests show a 3x performance enhancement. Some last changes are required for OpenTSDB. I will release the new code tomorrow.

**Comment by csoulios** [comment]
Released version v0.3.0 and submitted pull request to OpenTSDB

**Comment by mbrukman** [comment]
For those following along, https://github.com/OpenTSDB/opentsdb/pull/972 updates the Bigtable dependency in OpenTSDB core.
